src/components/input-number/input-number.tsx

```diff
@@ -0,0 +1,884 @@
+    const loader = (
+      <div class={CSS.loader}>
+        <calcite-progress label={this.intlLoading} type="indeterminate" />
+      </div>
+    );
+
+    const inputClearButton = (
+      <button
+        aria-label={this.intlClear || TEXT.clear}
+        class={CSS.clearButton}
+        disabled={this.disabled || this.readOnly}
+        onClick={this.clearInputValue}
+        tabIndex={this.disabled ? -1 : 0}
+        type="button"
+      >
+        <calcite-icon icon="x" scale="s" />
+      </button>
+    );
+    const iconEl = (
+      <calcite-icon
+        class={CSS.inputIcon}
+        flipRtl={this.iconFlipRtl}
+        icon={this.requestedIcon}
+        scale="s"
+      />
+    );
+
+    const isHorizontalNumberButton = this.numberButtonType === "horizontal";
+
+    const numberButtonsHorizontalUp = (
+      <button
+        class={{
+          [CSS.numberButtonItem]: true,
+          [CSS.buttonItemHorizontal]: isHorizontalNumberButton
+        }}
+        data-adjustment="up"
+        disabled={this.disabled || this.readOnly}
+        onPointerDown={this.numberButtonPointerDownHandler}
+        onPointerOut={this.numberButtonPointerUpAndOutHandler}
+        onPointerUp={this.numberButtonPointerUpAndOutHandler}
+        tabIndex={-1}
+        type="button"
+      >
+        <calcite-icon icon="chevron-up" scale="s" />
+      </button>
+    );
+
+    const numberButtonsHorizontalDown = (
+      <button
+        class={{
+          [CSS.numberButtonItem]: true,
+          [CSS.buttonItemHorizontal]: isHorizontalNumberButton
+        }}
+        data-adjustment="down"
+        disabled={this.disabled || this.readOnly}
+        onPointerDown={this.numberButtonPointerDownHandler}
+        onPointerOut={this.numberButtonPointerUpAndOutHandler}
+        onPointerUp={this.numberButtonPointerUpAndOutHandler}
+        tabIndex={-1}
+        type="button"
+      >
+        <calcite-icon icon="chevron-down" scale="s" />
+      </button>
+    );
+
+    const numberButtonsVertical = (
+      <div class={CSS.numberButtonWrapper}>
+        {numberButtonsHorizontalUp}
+        {numberButtonsHorizontalDown}
+      </div>
+    );
+
+    const prefixText = <div class={CSS.prefix}>{this.prefixText}</div>;
+
+    const suffixText = <div class={CSS.suffix}>{this.suffixText}</div>;
+
+    const childEl = (
+      <input
+        aria-label={getLabelText(this)}
+        autofocus={this.autofocus ? true : null}
+        defaultValue={this.defaultValue}
+        disabled={this.disabled ? true : null}
+        enterKeyHint={this.el.enterKeyHint}
+        inputMode={this.el.inputMode}
+        key="localized-input"
+        maxLength={this.maxLength}
+        minLength={this.minLength}
+        name={undefined}
+        onBlur={this.inputBlurHandler}
+        onFocus={this.inputFocusHandler}
+        onInput={this.inputNumberInputHandler}
+        onKeyDown={this.inputNumberKeyDownHandler}
+        onKeyUp={this.inputKeyUpHandler}
```

Comment:
The naming here is inconsistent.  Since there is only a single `childEl` being rendered, we could get away with removing the word `Number` from the names, especially since this renders as a `type="text"` anyway for l10n support.  I can see leaving the word `Number` in the name confusing to newcomers who might think that it's referring to a `type="number"` input element.

Reply:
I'd honestly go the other direction so that they all have `Number`. It would make searching the codebase easier when the different inputs have different variable names. It would also match the events which have `Number` in the name. But I do agree they should be consistent.

Reply:
Yeah, I like that.  I was at first thinking because these are internal methods they should be named specifically for the type of input they are being attached to, but I like that it would make it easier to search the codebase for handler functions.